### PR TITLE
[4328][ADD] report_qweb_decimal_precision

### PR DESCRIPTION
**Pull request by liuhehe1995**
[4328](https://www.quartile.co/web#id=4328&menu_id=505&cids=3&action=1457&model=project.task&view_type=form)

This module allows administrators to define the decimal precision of float fields for QWeb report presentation.

### CODE REVIEW

**Comment by kanda999** [comment]
Could you please fix pre-commit error?  
Basically, you request a review after the errors are resolved.

File: report_qweb_decimal_precision/i18n/ja.po [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+#. module: report_qweb_decimal_precision
+#: model:ir.model.fields,field_description:report_qweb_decimal_precision.field_decimal_precision_qweb__currency_id
+msgid "Currency"
+msgstr "通貨"
+
+#. module: report_qweb_decimal_precision
+#: model:ir.model.fields,field_description:report_qweb_decimal_precision.field_decimal_precision_qweb__currency_field_id
+msgid "Currency Field"
+msgstr "通貨フィールド"
```
Comment (yostashiro) [inline report_qweb_decimal_precision/i18n/ja.po]:
```suggestion
msgstr "通貨項目"
```

File: report_qweb_decimal_precision/i18n/ja.po [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+#. module: report_qweb_decimal_precision
+#: model:ir.actions.act_window,name:report_qweb_decimal_precision.action_decimal_precision_qweb_view
+#: model:ir.ui.menu,name:report_qweb_decimal_precision.menu_decimal_precision_qweb
+msgid "Decimal Precision QWeb"
+msgstr "小数点精度QWeb"
```
Comment (yostashiro) [inline report_qweb_decimal_precision/i18n/ja.po]:
```suggestion
msgstr "QWeb小数点精度"
```

File: report_qweb_decimal_precision/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2024 Quartile Limited (https://www.quartile.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Report Qweb Decimal Precision",
```
Comment (yostashiro) [inline report_qweb_decimal_precision/__manifest__.py]:
```suggestion
    "name": "Report QWeb Decimal Precision",
```

File: report_qweb_decimal_precision/i18n/ja.po [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+#. module: report_qweb_decimal_precision
+#: model:ir.model.fields,field_description:report_qweb_decimal_precision.field_decimal_precision_qweb__display_name
+msgid "Display Name"
+msgstr "表示名"
+
+#. module: report_qweb_decimal_precision
+#: model:ir.model.fields,field_description:report_qweb_decimal_precision.field_decimal_precision_qweb__field_id
+msgid "Field"
+msgstr "フィールド"
```
Comment (yostashiro) [inline report_qweb_decimal_precision/i18n/ja.po]:
```suggestion
msgstr "項目"
```

File: report_qweb_decimal_precision/i18n/ja.po [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+#. module: report_qweb_decimal_precision
+#: model:ir.model.fields,field_description:report_qweb_decimal_precision.field_decimal_precision_qweb__field_name
+msgid "Field Name"
+msgstr "フィールド名"
```
Comment (yostashiro) [inline report_qweb_decimal_precision/i18n/ja.po]:
```suggestion
msgstr "項目名"
```

File: report_qweb_decimal_precision/i18n/ja.po [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+#. module: report_qweb_decimal_precision
+#: model:ir.model.fields,field_description:report_qweb_decimal_precision.field_decimal_precision_qweb__res_model_name
+msgid "Model Name"
+msgstr "モデル名"
+
+#. module: report_qweb_decimal_precision
+#: model_terms:ir.ui.view,arch_db:report_qweb_decimal_precision.view_decimal_precision_qweb_searh
+msgid "Precision QWeb Search"
+msgstr "QWeb精密検索"
```
Comment (yostashiro) [inline report_qweb_decimal_precision/i18n/ja.po]:
```suggestion
msgstr "QWeb精度検索"
```

File: report_qweb_decimal_precision/i18n/ja.po [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+#. module: report_qweb_decimal_precision
+#: model:ir.model,name:report_qweb_decimal_precision.model_ir_qweb_field_float
+msgid "Qweb Field Float"
+msgstr "Qweb浮動小数点フィールド"
```
Comment (yostashiro) [inline report_qweb_decimal_precision/i18n/ja.po]:
```suggestion
msgstr "QＷeb浮動小数点項目"
```

File: report_qweb_decimal_precision/i18n/ja.po [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+#. module: report_qweb_decimal_precision
+#: model:ir.model.fields,field_description:report_qweb_decimal_precision.field_decimal_precision_qweb__uom_id
+msgid "UoM"
+msgstr "計量単位"
+
+#. module: report_qweb_decimal_precision
+#: model:ir.model.fields,field_description:report_qweb_decimal_precision.field_decimal_precision_qweb__uom_field_id
+msgid "UoM Field"
+msgstr "計量単位フィールド"
```
Comment (yostashiro) [inline report_qweb_decimal_precision/i18n/ja.po]:
```suggestion
msgstr "計量単位項目"
```

File: report_qweb_decimal_precision/__manifest__.py (hunk 3 above)
Comment (yostashiro) [reply report_qweb_decimal_precision/__manifest__.py]: I've updated the OCA PR on this point.

**Review by yostashiro — COMMENTED** [review]
@liuhehe1995 Please regenerate readme and translation file.

File: report_qweb_decimal_precision/models/decimal_precision_qweb.py [hunk 9]
```diff
@@ -0,0 +1,59 @@
+# Copyright 2024 Quartile Limited (https://www.quartile.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class DecimalPrecisioQweb(models.Model):
+    _name = "decimal.precision.qweb"
+    _description = "Decimal Precision Qweb"
```
Comment (yostashiro) [inline report_qweb_decimal_precision/models/decimal_precision_qweb.py]:
```suggestion
    _description = "Decimal Precision QWeb"
```

**Review by yostashiro — APPROVED** [review]

**Pull request merged**